src/core/Table/Table.tsx

```diff
@@ -278,13 +291,16 @@ export const Table = <
       defaultColumn,
       disableSortBy: !isSortable,
       stateReducer: tableStateReducer,
-      filterTypes: { ...customFilterFunctions, ...filterFunctions },
+      filterTypes,
+      selectSubRows,
     },
     useFlexLayout,
     useFilters,
+    useSubRowFiltering(showSubRows),
     useSortBy,
     useExpanded,
     useRowSelect,
+    useSubRowSelection,
     useExpanderCell(subComponent, expanderCell, isRowDisabled),
     useSelectionCell(isSelectable, isRowDisabled),
```

Comment:
There is an issue with filtered sub-rows selection when selecting parent row and selecting again. This is caused that nothing really changed for the row and it is not being rerendered but checkbox does what it has to do - it unmarks indeterminate state. 
![table-subrows-filtered-selection-bug](https://user-images.githubusercontent.com/36186912/129683090-70b881a2-6639-4212-bd1e-6668402fb6fd.gif)

The only workaround that I can think of is making checkbox cell to rerender:
![image](https://user-images.githubusercontent.com/36186912/129683241-c7104853-fd09-4e58-a9fe-583f53d1cfb1.png)

Reply:
I think it should mark to checked, when you click..

Reply:
why mark to checked? some rows are hidden

Reply:
clickint on a checkbox to get indetermiante state is just wrong (not valid). Indeterminate state cannot be achieved interactively by design.

Reply:
I did the same as we are handling simple filtered rows. And previously we were discussing that it should show indeterminate state, as react-table was showing checked. So I don't know if it is correct to show checked here. And maybe you have any link about showing indeterminate being a bad practice?

Reply:
Well, indeterminate can be achieved only through js. So user click on a checkbox to make it indeterminate looks weird. There is css-tricks article, where they made rotating checkbox UI example...https://css-tricks.com/indeterminate-checkboxes/
I guess the only argument is that if user clicks and gets indetermiante, it is not clear, what is actually selected, what is not. Cause common UX is when you click, you check or uncheck.
Dont know... But in that GIF, if you uncheck indeterminate, nothing really changes in children selection (they should go unchecked..?).

I guess need to try to find same examples in other libs and see how they handle..

Reply:
Or we could change how our checkbox works: instead of going from indeterminate to true, it should go from indeterminate to false.

Reply:
thats a big one...

Reply:
Created an issue for this: https://github.com/iTwin/iTwinUI-react/issues/271